src/widgets/query-rule-custom-data/query-rule-custom-data.tsx

```diff
@@ -52,9 +57,11 @@ const renderer = ({
   containerNode: HTMLElement;
   cssClasses: QueryRuleCustomDataComponentCSSClasses;
   renderState: {
-    templateProps?: PreparedTemplateProps<QueryRuleCustomDataTemplates>;
+    templateProps?: PreparedTemplateProps<
+      QueryRuleCustomDataComponentTemplates
+    >;
```

Comment:
renderState is defined, but we don't use it, and rely on spreading in the widget itself?